### PR TITLE
Add new help messages for E0425

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Last part of #30413.

r? @pnkfelix

### CODE REVIEW

**Comment by pnkfelix** [comment]
@GuillaumeGomez where are the tests of the new messages you are adding here?

see e.g. https://github.com/pnkfelix/rust/commit/694699503a346822ae8453e93c9f327d01fe6f55 from #30413

**Comment by GuillaumeGomez** [comment]
Tests have been added.

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -444,9 +443,15 @@ fn resolve_error<'b, 'a: 'b, 'tcx: 'a>(resolver: &'b Resolver<'a, 'tcx>,
...
-
-                            _ => {} // no help available
+                            ExprCall(_, _) => {
+                                help_msg = format!("No function corresponds to `{module}(..)`",
+                                                   module = &*path);
+                            }
+                            _ => { } // no help available
                         }
+                    } else {
+                        help_msg = format!("Module `{module}` cannot be used as an object!",
```
Comment (pnkfelix) [inline src/librustc_resolve/lib.rs]: hmm, do we used the terminology "used as an object" elsewhere?  (That is supposed to mean that its been used as the receiver in a method call, right?)

On the one hand, your phrasing is succinct. But on the other hand, I'm not sure everyone will understand what it means ...

Comment (pnkfelix) [inline src/librustc_resolve/lib.rs]: oh, duh, I should have looked at the test!

Comment (pnkfelix) [inline src/librustc_resolve/lib.rs]: Okay, so I *don't* like the phrasing "used as an object" here.

I would instead say something like "Module `{module}` cannot be passed around as a first-class value."

(except that may be a bit long winded... maybe just "Module `{module}` cannot be the value of an expression" ?)

**Comment by GuillaumeGomez** [comment]
I changed the message, tell me if it fits better what you have in mind.

**Comment by pnkfelix** [comment]
@GuillaumeGomez thanks, sorry for the slow processing of the PR!

**Pull request merged**